test/system/030-run.bats

```diff
@@ -758,4 +758,9 @@ EOF
     is "$output" ".*TERM=abc" "missing TERM environment variable despite TERM being set on commandline"
 }
 
+@test "podman run --kernel-memory warning" {
+    # Not sure what situations this fails in, but want to make sure warning shows.
+    run_podman '?' run --rm --kernel-memory 100 $IMAGE false
+    is "$output" ".*The --kernel-memory flag has been deprecated. May not work properly on your system." "warn on use of --kernel-memory"
+}
```

Comment:
This test timed out in CI???

Reply:
...on f34, which uses cgroupsv1, which is exactly the kind of setup I don't have access to. Does anyone have a cgroupsv1 host they can test this on? If not I'll try to make one tomorrow.

Reply:
Confirmed: this command cannot be run on a cgroupsv1 system:
```console
$ podman run --rm --kernel-memory 100 quay.io/libpod/testimage:20210610 false
```
It hangs, and also hangs `podman ps` and `podman info` (some lock, no doubt).

I have a 1minutetip VM exhibiting this. @mheon, want it?

And, for future reference, the magic incantation to enable cgroupsv1 in 1minutetip is:
```console
# grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
```
...then log out and use `,` (comma) to reboot.

Reply:
Did you see the warning? Perhaps we should just throw an error.

Reply:
I'm running podman from RPM, not compiled. Compiling podman on 1minutetip is too much of a hassle; all I wanted to do was confirm the hang (and the reason the test is failing).

I kind of don't like the situation where a magic podman command breaks everything for everybody, and will probably file an issue.

Reply:
I have just blown away the option altogether. Now it is just a NOOP.

Reply:
isn't the value in bytes?  With such low values the OCI runtime dies before it even gets to run the container payload.  I think we should try at least with ~10 to be safe with runc